src/java.base/share/classes/sun/net/www/http/HttpClient.java

```diff
@@ -1159,6 +1159,8 @@ public void setIgnoreContinue(boolean value) {
     public void closeServer() {
         try {
             keepingAlive = false;
+            // SSLSocket.close may block up to timeout. Make sure it's short.
+            serverSocket.setSoTimeout(1);
```

Comment:
maybe we should try catch and ignore exception on this line - or close the socket on a finally block. An alternative could be to only do this for SSLSocket - in the HttpsClient - by overriding closeServer() there. Also I see that there are other places where serverSocket.close() is called in the HttpsClient.